lib/matplotlib/axes/_base.py

```diff
@@ -4123,9 +4134,102 @@ def twiny(self):
         return ax2
 
     def get_shared_x_axes(self):
-        'Return a copy of the shared axes Grouper object for x axes'
+        """Return a copy of the shared axes Weakset object for x axes"""
         return self._shared_x_axes
 
     def get_shared_y_axes(self):
-        'Return a copy of the shared axes Grouper object for y axes'
+        """Return a copy of the shared axes Weakset object for y axes"""
         return self._shared_y_axes
+
+    def unshare_x_axes(self, axes=None):
+        """
+        Unshare x axis.
+
+        Parameters
+        ----------
+        axes: Axes
+            Axes to unshare. None will unshare itself.
+        """
+        if axes is None or axes is self:
+            self._shared_x_axes.remove(self)
+            self._shared_x_axes = WeakSet()
+            self._shared_x_axes.add(self)
+        elif axes in self._shared_x_axes:
+            axes.unshare_x_axes()
+
+    def unshare_y_axes(self, axes=None):
+        """
+        Unshare y axis.
+
+        Parameters
+        ----------
+        axes: Axes
+            Axes to unshare. None will unshare itself.
+        """
+        if axes is None or axes is self:
+            self._shared_x_axes.remove(self)
+            self._shared_x_axes = WeakSet()
+            self._shared_x_axes.add(self)
+        elif axes in self._shared_y_axes:
+            axes.unshare_y_axes()
+
+    def unshare_axes(self, axes=None):
+        """
+        Unshare both x and y axis.
+
+        Parameters
+        ----------
+        axes: Axes
+            Axes to unshare. None will unshare itself.
+        """
+        self.unshare_x_axes(axes)
+        self.unshare_y_axes(axes)
+
+    def share_x_axes(self, axes):
+        """
+        Share x axis
+        Parameters
+        ----------
+        axes: Axes
+            Axes to share.
+        """
+        self._shared_x_axes |= axes._shared_x_axes
+        # Transitive, propagation to other axes.
+        for ax in self._shared_x_axes:
+            ax._shared_x_axes = self._shared_x_axes
+            if ax._adjustable == 'box':
+                ax._adjustable = 'datalim'
+
+        # keep original intention
+        axes._adjustable = 'datalim'
+        self._adjustable = 'datalim'
```

Comment:
What happens if one calls set_adjustable() on an axes which is shared with others?  Should that be prevented, or somehow propagated?

Reply:
It is a preliminary pr. Discussion is needed. See there are some build error related to this set_adjustable()

Reply:
This is a messy question partly addressed by, or at least related to #8752, which needs work to handle twinx, twiny properly and to remove some parts that are no longer needed because of #8626.